src/app/_mermaid_common/libs/directives/mapbox-gl.dir.js

```diff
@@ -79,55 +122,199 @@ angular.module('mermaid.libs').directive('mapboxGl', [
+                1,
+                ['==', ['get', 'geomorphic'], 'Small Reef'],
+                1,
+                ['==', ['get', 'geomorphic'], 'Terrestrial Reef Flat'],
+                1,
+                ['==', ['get', 'geomorphic'], 'Unknown'],
+                1,
+                0 // Default / other
+              ];
+        scope.rasterOpacityExpression =
+          rasterOpacityValue === 0 || rasterOpacityValue
+            ? rasterOpacityValue
+            : 1;
+
         const defaultCenter = scope.mapopts.defaultCenter || [20, 0.0];
         const defaultZoom = scope.mapopts.defaultZoom || 1;
         const popup = scope.mapopts.popup || false;
+        const slider = scope.mapopts.slider || false;
         const navigation = new mapboxgl.NavigationControl({
           showCompass: false,
           showZoom: true
         });
 
-        const mapBox = new mapboxgl.Map({
+        const map = new mapboxgl.Map({
           container: 'map',
           style: worldBaseMap,
           center: defaultCenter, // starting position [lng, lat]
           zoom: defaultZoom, // starting zoom
           maxZoom: 16
         });
 
-        mapBox.addControl(navigation, 'top-left');
-
-        mapBox.scrollZoom.disable();
-        mapBox.dragRotate.disable();
-        mapBox.touchZoomRotate.disableRotation();
-        mapBox.scrollZoom.setWheelZoomRate(0.02); // Default 1/450
-
-        mapBox.on('wheel', event => {
-          if (event.originalEvent.ctrlKey) {
-            event.originalEvent.preventDefault();
-            hideHelp(mapBox);
-            if (!mapBox.scrollZoom._enabled) mapBox.scrollZoom.enable();
-          } else {
-            if (mapBox.scrollZoom._enabled) {
-              mapBox.scrollZoom.disable();
-            }
-            showHelp(mapBox, settings.textMessage);
-            setTimeout(() => {
-              hideHelp(mapBox);
-            }, settings.timeout);
-          }
-        });
-
-        mapBox.on('load', function() {
-          mapBox.addSource('mapMarkers', {
+        const initLoadMapLayers = function() {
+          map.addSource('mapMarkers', {
             type: 'geojson',
             data: {
               type: 'FeatureCollection',
               features: []
             }
           });
 
-          mapBox.addLayer({
+          map.addSource('atlas-planet', {
+            type: 'raster',
+            tiles: [
+              'https://integration.allencoralatlas.org/tiles/planet/visual/2019/{z}/{x}/{y}'
+            ]
+          });
+
+          map.addSource('atlas-benthic', {
+            type: 'vector',
+            tiles: [
+              // 'http://34.83.20.4:8080/geoserver/gwc/service/tms/1.0.0/coral-atlas:reef_polygons_benthic_expanded@EPSG:900913@pbf/{z}/{x}/{y}.pbf'
+              'https://integration.allencoralatlas.org/geoserver/gwc/service/tms/1.0.0/coral-atlas:reef_polygons_benthic_expanded@EPSG:900913@pbf/{z}/{x}/{y}.pbf '
```

Comment:
In Vulcan's documentation, they said:
> Tile server vector tile endpoints may be more desirable than using geoserver for vector tiles due to caching. **Tile generation with geoserver due to empty cache may be unbearably slow.
Did you try those? I do find the tiles pretty slow... I think they mean the endpoints like `https://integration.allencoralatlas.org/tiles/benthic/{z}/{x}/{y}` but am not sure.